executor/src/core/agent/Auth.ts

```diff
@@ -351,6 +353,25 @@ export const RUNTIME_EXCEPTION_SUBSCRIBE_CAPABILITY: Capability = {
     can: [SUBSCRIBE]
 }
 
+export const checkTokenAuthorized = (rootConfigPath: string, token: string, isAd4minCredential: boolean) => {
+    let apps: any[] = [];
+    if (!isAd4minCredential) {
+        if (fs.existsSync(path.join(rootConfigPath, "apps.json"))) {
+            apps = JSON.parse(fs.readFileSync(path.join(rootConfigPath, "apps.json")).toString())
```

Comment:
I'm a bit worried that this file access can slow things down since this function gets called in every interface query.  Might make sense to hold the `apps` array in memory as well?